crates/turbopath/src/relative_unix_path.rs

```diff
@@ -0,0 +1,21 @@
+use std::path::PathBuf;
+
+use path_slash::PathBufExt;
+
+use crate::RelativeSystemPathBuf;
+
+pub struct RelativeUnixPath<'a> {
```

Comment:
So the reason we only have `RelativeSystemPathBuf` and not `RelativeSystemPath` in turbopath is because in the process of creating the path, there is a chance we need to convert the slashes, which requires allocation. Since the `-Path` types are unsized and therefore not an allocation, we can't create them without throwing out automatic conversion.

To sum it up, if we want a `RelativeUnixPath` type, we either need to validate but not convert on construction, or we need to make this only constructible via a `RelativeUnixPathBuf`

Reply:
Yeah we'll probably want both at some point, also because `&RelativeSystemPathBuf` is a little inefficient (similar to `&String` versus `&str`). If we want the full infrastructure, we'd have `RelativeSystemPathBuf::new` return a `Cow` that is either a `RelativeSystemPath` or a `RelativeSystemPathBuf` depending on whether we allocate or not. That'd be the most idiomatic Rust version. But I think for now we should stick with `RelativeSystemPathBuf`

Reply:
I reworked `RelativeUnixPath` to be based on the `Path` implementation in the stdlib. Seems ok so far.